docs/documentation-generation/01-overview.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Stencil Documentation Generation
+sidebar_label: Overview
+description: Stencil Documentation Generation
+slug: /doc-generation
+---
+
+# Documentation Generation
+
+- [`docs-readme`: Documentation readme files formatted in markdown](./docs-readme.md)
+- [`docs-json`: Documentation data formatted in JSON](./docs-json.md)
+- [`docs-custom`: Custom documentation generation](./docs-custom.md)
+- [`docs-vscode`: Documentation generation for VS Code](./docs-vscode.md)
+- [`stats`: Stats about the compiled files](./docs-stats.md)
+
+## Docs Auto-Generation
+
+As apps scale with more and more components, and team size and members continue to adjust over time, it's vital all components are well documented, and that the documentation itself is maintained. Maintaining documentation is right up there with some of the least interesting things developers must do, but that doesn't mean it can't be made easier.
```

Comment:
I sort of like writing documentation / technical things! It's a great way to clarify ideas and present how something works - I'd kind of like it if we could remove / change the wording here that suggests it's boring or a chore to write documentation — I know many developers do feel that way though, so maybe we could say something more like

"Whether you love taking the time to write documentation or you can't wait to get coding again, we could all use some help in keeping documentation up-to-date when we change components and APIs. Stencil provides some built-in tools to automate extracting information from your component library, providing flexible output formats that you can use when building out your library's documentation."

Idk if that quite rolls of the tongue, but something along those lines?